### ISSUE TITLE
Label for should apply to contenteditable elements

### ISSUE THREAD

**Issue by darobin** [w3c/html#734]
In many ways, `contenteditable` elements are comparable to somewhat rich textareas. It would be useful for them to be labelled using `<label for="...">`. Right now https://www.w3.org/TR/html51/sec-forms.html#labelable-element says no.

**Comment by stevefaulkner** [comment]
@darobin do you mean be allowed (as conforming) or actually work like a label does i.e. move focus and provide an accessible name for the contenteditbale element. The latter involves browsers implementing...

**Comment by chaals** [comment]
It seems like it should **work** - apart from making it possible to test out in JS with valid HTML, is there any real benefit to just allowing it?

**Comment by stevefaulkner** [comment]
FYI, knocked up an example of how it could/should work
http://codepen.io/stevef/pen/Obpxbd

**Comment by chaals** [comment]
Nice. What happens in forms mode in screenreaders, using the example?

**Comment by stevefaulkner** [comment]
@chaals, works as expected with JAWS, haven't tried any others

**Comment by LJWatson** [comment]
Using Steve's example... Jaws 18 supports it in FF, Chrome and IE (without relying on its label heuristic peculiarities). Narrator supports it in IE but not Edge. NVDA doesn't seem to support it in FF, Chrome or IE.

**Comment by ZoeBijl** [comment]
Works as expected in TalkBack with Firefox on Android 7.

**Comment by chaals** [comment]
By which you mean that it lets you navigate to the label, and that activates the editable region, and that navigating to the editable region reads out the label?23.11.2016, 15:58, "Michiel Bijl" notifications@github.com:Works as expected in TalkBack with Firefox on Android 7.

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.

-- Charles McCathie Nevile - standards - Yandexchaals@yandex-team.ru - - - Find more at http://yandex.com

**Comment by ZoeBijl** [comment]
@chaals yes.

**Comment by LJWatson** [comment]
It is worth noting that on most touch devices running a screen reader, there is no concept of navigating to the label (as opposed to the field it's associated with). Navigation is to the field directly, and the label and field are effectively an atomic object.

**Comment by chaals** [comment]
Did anyone write a test that just puts a label on a contenteditable element? I assumed that it wouldn't work, but testing it seems sensible…

**Comment by LJWatson** [comment]
@chaals @stevefaulkner  posted <a href="http://codepen.io/stevef/pen/Obpxbd">an example</a> in an <a href="https://github.com/w3c/html/issues/734#issuecomment-261995346">earlier comment</a>.

**Comment by ZoeBijl** [comment]
@chaals here is a [demo and test results without JS or ARIA](https://moiety.me/research/contenteditable/), in short: only works in IE 11.

@LJWatson think Chaals meant a demo without the added JS that Steve put in there to make clicking the label work.

**Comment by LJWatson** [comment]
Retesting this, six months on from @MichielBijl's original tests.

Test criteria:
* Clicking on the ```<label>``` activates the contenteditable area;
* The content of the ```<label>``` is exposed as the contenteditable area's accessible name.

It seems that only IE meets either of these criteria. Chrome, Firefox, Edge, and Safari do not meet either criteria.

It seems unlikely that implementors will run with this, without a groundswell of support, polyfills, and use in the wild.

The desired behaviour can be polyfilled with ARIA, but that only meets the second of the criteria of course.

**Comment by ZoeBijl** [comment]
@LJWatson you can still add a click event handler to the label and make that activate the contenteditable area. In other words, you can polyfill both criteria, no?

**Comment by LJWatson** [comment]
Good point @MichielBijl. I don't think it changes the basic problem though - that without implementations, we can't add this to the spec.

**Comment by ZoeBijl** [comment]
@ljwatson agreed.

I’m not aware of any such implementations in the wild. Twitter’s compose box uses contenteditable but doesn’t have a visible label, let alone one that’s clickable. GitHub and LinkedIn both use a `<textarea>` for their main comment/content submission forms.

**Comment by scottaohara** [comment]
Testing again almost a year later, the only changes are:

Windows 10 + IE11
Label announced as accessible name with JAWS2018, but not NVDA 2018.1. Still can't click to activate.

Windows 10 + Chrome 66
Label announced as accessible name with JAWS2018, but not NVDA 2018.1. Still can't click to activate.

**Comment by siusin** [comment]
Thanks all.

We're closing this issue on the W3C HTML specification because the W3C and WHATWG are now [working together on HTML](https://www.w3.org/blog/2019/05/w3c-and-whatwg-to-work-together-to-advance-the-open-web-platform/), and all issues are being discussed on the [WHATWG repository](https://github.com/whatwg/html).

If you filed this issue and you still think it is relevant, please open a [new issue](https://github.com/whatwg/html/issues) on the WHATWG repository and reference this issue (if there is useful information here). Before you open a new issue, please check for existing issues on the WHATWG repository to avoid duplication.

If you have questions about this, please open an issue on the [W3C HTML WG repository](https://github.com/w3c/htmlwg/issues/new) or send an email to public-html@w3.org.
